src/main/java/com/exactpro/th2/sim/run/SimulatorServerMain.java

```diff
@@ -43,11 +43,7 @@ public static void main(String[] args) {
 
     private static void addShutdownHook(SimulatorServer server, CommonFactory factory) {
         Runtime.getRuntime().addShutdownHook(new Thread(() -> {
-            try {
-                try (factory) {
-                    server.close();
-                }
-            } catch (Exception e) {
+            try (factory; server) {} catch (Exception e) {
                 LOGGER.error(e.getMessage(), e);
             }
```

Comment:
```suggestion
            try {
                try (factory; server) {}
            } catch (Exception e) {
                LOGGER.error(e.getMessage(), e);
            }
```
Because I doubt that catch-block will be triggered by exceptions thrown during closing of resources